presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -257,7 +259,11 @@ private Date getDate(int columnIndex, DateTimeZone localTimeZone)
         }
 
         try {
-            return new Date(DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value)));
+            LocalDate localDate = DATE_FORMATTER.parseLocalDate(String.valueOf(value));
+            GregorianCalendar calendar = new GregorianCalendar(localDate.getYear(), localDate.getMonthOfYear() - 1, localDate.getDayOfMonth());
```

Comment:
This seems fairly expensive. Can we only do this for dates in the affected range, and simply use `LocalDate` otherwise?

Reply:
I thought about it, because I had heard complaints about performance, though I have never benchmarked myself or seen a real benchmark.  But it's easy enough to make this conditional.

Reply:
By a rule of thumb, we  should stay away from Calendar unless benchmarks provide assurance it's OK to use it.
(of course, this applies to "reasonable dates", I am fine with using Calendar for historical dates if needed)

Reply:
I ran a simple micro benchmark and the approach using `GregorianCalendar` appears indeed to be much slower than Joda.

Reply:
> I ran a simple micro benchmark a

Is it in a shape that we could add to the project?
Using an in memory result set we should be able to 
abstract the network communication part.